### DISCUSSION TITLE
Unusual if statement in MaterialDesign codebase - would appreciate help

### DISCUSSION THREAD

**Discussion by sabuncumurat** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3339] [Q&A] [answered]
Hi, when browsing the codebase, I came across the following if statement:

![2023-10-14 213156 - devenv](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/91539510/cdaf3284-86ec-4ed2-b33e-ebb6fc2b88f5)

I have not kept up with changes in the C# language, would appreciate if you could explain how the braces and the variable work.

And most importantly, what is the official name of this construct in C# standard?

**Comment by Keboo** [comment, marked as answer]
The name for this is "property pattern matching". It is both performing a null check, but if the value of `HintProxy` is not null, it will assign the non-null value to `hintProxy` and the expression returns true. 

You can read more about it here: https://essentialcsharp.com/pattern-matching#property-patterns-c-8-0-10-0

**Reply by sabuncumurat** [reply]
Thank you so much for this.
Regards.
